services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/hakemus/HakemusService.kt

```diff
@@ -258,6 +283,54 @@ class HakemusService(
         alluStatusRepository.save(status)
     }
 
+    private fun newApplicationData(createHakemusRequest: CreateHakemusRequest): ApplicationData =
+        when (createHakemusRequest) {
+            is CreateJohtoselvityshakemusRequest ->
+                createHakemusRequest.newCableReportApplicationData()
+            is CreateKaivuilmoitusRequest -> createHakemusRequest.newExcavationNotificationData()
+        }
+
+    private fun CreateJohtoselvityshakemusRequest.newCableReportApplicationData() =
```

Comment:
Since these extension functions are private and only used in this service, they could also be regular private functions, e.g. `fun newCableReportApplicationData(request: CreateJohtoselvityshakemusRequest)`. But I guess it does not really matter?

Reply:
Then I would have to add `request.` to any values copied from the request.

Reply:
Good point!